### PR TITLE
Tighten pure config validation [DPP-1205]

### PR DESCRIPTION
**Pull request by skisel-da**
CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: ledger/ledger-runner-common/src/main/scala/com/daml/ledger/runner/common/PureConfigReaderWriter.scala [hunk 1]
```diff
@@ -228,6 +238,9 @@ class PureConfigReaderWriter(secure: Boolean = true) {
 
   implicit val seedingWriter: ConfigWriter[Seeding] = ConfigWriter.toString(_.name)
 
+  implicit val userManagementConfigHint =
+    ProductHint[UserManagementConfig](allowUnknownKeys = false)
+
```
Comment (mziolekda) [inline ledger/ledger-runner-common/src/main/scala/com/daml/ledger/runner/common/PureConfigReaderWriter.scala:243]: Seems `jwtTimestampLeewayConfigConvert` below is missing a hint

Comment (skisel-da) [reply ledger/ledger-runner-common/src/main/scala/com/daml/ledger/runner/common/PureConfigReaderWriter.scala:243]: ah actually all optional configs, let me fix that

Comment (skisel-da) [reply ledger/ledger-runner-common/src/main/scala/com/daml/ledger/runner/common/PureConfigReaderWriter.scala:198]: done

**Review by mziolekda — APPROVED** [review]

File: ledger/ledger-runner-common/src/test/scala/com/daml/ledger/runner/common/PureConfigReaderWriterSpec.scala [hunk 2]
```diff
@@ -189,6 +196,14 @@ class PureConfigReaderWriterSpec
     )
   }
 
+  it should "not support unknown keys" in {
+    convert(
+      jwtTimestampLeewayConfigConvert,
+      "some-crap=yes\n" + validJwtTimestampLeewayValue,
```
Comment (miklos-da) [inline ledger/ledger-runner-common/src/test/scala/com/daml/ledger/runner/common/PureConfigReaderWriterSpec.scala:202]: I'd suggest the key `some-unknown-key` instead ; )

Comment (mziolekda) [reply ledger/ledger-runner-common/src/test/scala/com/daml/ledger/runner/common/PureConfigReaderWriterSpec.scala:202]: agreed

**Pull request merged**